torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -1040,7 +1045,12 @@ class CudaKernelGenerator : private kir::IrVisitor {
   }
 
   void visit(const kir::Sync* node) final {
-    indent() << "__barrier_sync(0);\n";
+    // Use a custom synchronization method if enabled
```

Comment:
Why the guard here instead of always generating `block_sync::sync()`? That's what you're doing for all the headers right?

Reply:
No technical reason. I just didn't want to expose it to users if not necessary.